DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/PerturbProcessor.cs

```diff
@@ -0,0 +1,131 @@
+            { DicomVR.US, new VRTypes() { ElementType = typeof(DicomUnsignedShort), ValueType = typeof(ushort[]) } },
+            { DicomVR.OW, new VRTypes() { ElementType = typeof(DicomOtherWord), ValueType = typeof(ushort[]) } },
+            { DicomVR.UL, new VRTypes() { ElementType = typeof(DicomUnsignedLong), ValueType = typeof(uint[]) } },
+            { DicomVR.OL, new VRTypes() { ElementType = typeof(DicomOtherLong), ValueType = typeof(uint[]) } },
+            { DicomVR.UV, new VRTypes() { ElementType = typeof(DicomUnsignedVeryLong), ValueType = typeof(ulong[]) } },
+            { DicomVR.OV, new VRTypes() { ElementType = typeof(DicomOtherVeryLong), ValueType = typeof(ulong[]) } },
+            { DicomVR.SV, new VRTypes() { ElementType = typeof(DicomSignedVeryLong), ValueType = typeof(long[]) } },
+        };
+
+        public PerturbProcessor(IDicomAnonymizationSetting ruleSetting = null)
+        {
+            _perturbFunction = new PerturbFunction((DicomPerturbSetting)(ruleSetting ?? new DicomPerturbSetting()));
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            if (!IsValidItemForPerturb(item))
+            {
+                throw new AnonymizationOperationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationFunction, $"Perturb is not supported for {item.ValueRepresentation}");
+            }
+
+            if (item.ValueRepresentation == DicomVR.AS)
+            {
+                var values = ((DicomAgeString)item).Get<string[]>().Select(Utility.ParseAge).Select(x => _perturbFunction.Perturb(x));
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Select(Utility.AgeToString).ToArray());
+            }
+            else
+            {
+                var elementType = _numericValueTypeMapping[item.ValueRepresentation].ElementType;
+                var valueType = _numericValueTypeMapping[item.ValueRepresentation].ValueType;
+
+                // Get numeric value using reflection.
+                var valueObj = elementType.GetMethod("Get").MakeGenericMethod(valueType).Invoke(item, new object[] { -1 });
+                PerturbNumericValue(dicomDataset, item, valueObj as Array);
+            }
+        }
+
+        private void PerturbNumericValue(DicomDataset dicomDataset, DicomItem item, Array values)
+        {
+            if (values.Length == 0)
+            {
+                return;
+            }
+
+            var valueType = values.GetValue(0).GetType();
+            if (valueType == typeof(decimal))
+            {
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Cast<decimal>().Select(_perturbFunction.Perturb).ToArray());
```

Comment:
possible to use this interface AddOrUpdate(IEnumerable<DicomItem> items)?

Reply:
We still need to new items with different value types.